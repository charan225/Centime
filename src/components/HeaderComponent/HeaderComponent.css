.header-container{
    box-shadow: 0 2px 4px 0 rgb(0 0 0 / 10%);
    margin: 0!important;
    display: flex;
    align-items: center;
    background-color: #1da4a8;
    padding: 8px 24px 24px 16px;
    padding-bottom: 16px;    
    width: 100%;
}

.header-content{
    width: 100%;
}

.header-dropdown-container{
    width: 85%;
}

