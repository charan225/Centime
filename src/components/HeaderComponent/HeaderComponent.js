import { useEffect } from 'react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import LanguageIcon from '@material-ui/icons/Language';
import IconButton from "@material-ui/core/IconButton";

import logo from '../../assets/centimeLogo.png';
import {DropdownButton, Dropdown} from 'react-bootstrap';
import i18next from 'i18next';
import Cookies from 'js-cookie';
import { useDispatch } from 'react-redux';
import './HeaderComponent.css';

const languages = [
    {
        code: 'fr',
        name: 'Français',
        country_code: 'fr'
    },
    {
        code: 'en',
        name: 'English',
        country_code: 'gb'
    },
    {
        code: 'ar',
        name: 'عربى',
        country_code: 'sa',
        dir: 'rtl'
    }
]

const HeaderComponent = () => {   
    const [currentLanguageCode,setcurrentLanguageCode] = React.useState(Cookies.get('i18next') || 'en');
    const [currentLanguage, setcurrentLanguage] = React.useState(languages.find(l => l.code === currentLanguageCode));
    console.log(currentLanguageCode,currentLanguage, Cookies.get('i18Next'));
    
    useEffect(()=> {
        document.body.dir = currentLanguage.dir || "ltr";
        console.log("dir",document.body.dir);
    },[currentLanguageCode])

    const onChangeLang = (code)=>{
        i18next.changeLanguage(code);
        const newcurrentLanguageCode = Cookies.get('i18next') || 'en';
        const newCurrentLanguage = languages.find(l => l.code === newcurrentLanguageCode);
        console.log("onclick2",newCurrentLanguage);
        setcurrentLanguage(newCurrentLanguage);
        setcurrentLanguageCode(newcurrentLanguageCode);
    }
    const dropdownTitle = () => {
        return (
            <IconButton
                aria-label="select language" style={{ padding: 0 }}
            >
                <LanguageIcon style={{ fontSize: 24, padding: 0 , color: "white"}}/>
            </IconButton>
        )
    }
    return (
        <div className="header-container">
            <div className="display-flex header-content">
                <div className="float-left">
                    <img src={logo} width="210" height="40" alt="Centime"></img>
                </div>
                <div className="header-dropdown-container">
                    <div className="float-right">
                    <DropdownButton id="dropdown-basic-button" title={dropdownTitle()}>
                        {languages.map(({ code, name, country_code }) => (
                            <Dropdown.Item href={code} onClick={() => onChangeLang(code)}><span className={`m-r12 flag-icon flag-icon-${country_code}`}></span>
                                {name}</Dropdown.Item>

                        ))}
                    </DropdownButton>
                    </div>                    
                </div>
            </div>
        </div>
    )
}

export default HeaderComponent;