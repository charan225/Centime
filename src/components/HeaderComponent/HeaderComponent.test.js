import HeaderComponent from "./HeaderComponent";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

describe("Header component", () => {
  test("should render header correctly", () => {
    const wrapper = shallow(<HeaderComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
