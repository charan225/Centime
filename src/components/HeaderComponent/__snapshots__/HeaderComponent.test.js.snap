// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render header correctly 1`] = `
<div
  className="header-container"
>
  <div
    className="display-flex header-content"
  >
    <div
      className="float-left"
    >
      <img
        alt="Centime"
        height="40"
        src="centimeLogo.png"
        width="210"
      />
    </div>
    <div
      className="header-dropdown-container"
    >
      <div
        className="float-right"
      >
        <DropdownButton
          id="dropdown-basic-button"
          title={
            <WithStyles(ForwardRef(IconButton))
              aria-label="select language"
              style={
                Object {
                  "padding": 0,
                }
              }
            >
              <Memo
                style={
                  Object {
                    "color": "white",
                    "fontSize": 24,
                    "padding": 0,
                  }
                }
              />
            </WithStyles(ForwardRef(IconButton))>
          }
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            href="fr"
            onClick={[Function]}
          >
            <span
              className="m-r12 flag-icon flag-icon-fr"
            />
            Français
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            href="en"
            onClick={[Function]}
          >
            <span
              className="m-r12 flag-icon flag-icon-gb"
            />
            English
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            disabled={false}
            href="ar"
            onClick={[Function]}
          >
            <span
              className="m-r12 flag-icon flag-icon-sa"
            />
            عربى
          </DropdownItem>
        </DropdownButton>
      </div>
    </div>
  </div>
</div>
`;
